backvendor/workingtree.go

```diff
@@ -167,41 +167,66 @@ func (wt *WorkingTree) timeFromRevision(rev string) (time.Time, error) {
 	return t, err
 }
 
-// reachableTag returns the most recent reachable semver tag.
-func (wt *WorkingTree) reachableTag(rev string) (string, error) {
-	if wt.VCS.Cmd != vcsGit {
-		return "", ErrorUnknownVCS
+// gitReachableMatchingTag uses git to find a reachable tag from a
+// given revision rev. It limits the reachable tags using the glob
+// pattern match. If exact is true, the tag must be exactly at
+// revision rev, not just reachable (the output is parsed differently
+// in this case).
+func (wt *WorkingTree) gitReachableMatchingTag(rev, match string, exact bool) (string, error) {
+	argv := []string{"describe", "--tags", "--match=" + match}
+	if exact {
+		argv = append(argv, "--exact-match")
+	}
+	buf, err := wt.run(append(argv, rev)...)
+	output := strings.TrimSpace(buf.String())
+	if err != nil {
+		switch {
+		case strings.HasPrefix(output, "fatal: no tag exactly matches "),
```

Comment:
Optional: This seems a bit brittle, although I can't really think of a better approach. I assume these error messages are stable, but perhaps you could consider making the string lowercase to avoid changes in cases in the future. You could also just check if the message starts with `fatal: no tag` or `fatal: no names` or `fatal: no annotated tags`.

Reply:
Updated.